### PR TITLE
Add balance law to init funcs

### PR DESCRIPTION
**Pull request by charleskawczynski**
Adds balance law to initial condition functions, this will allow us to use `orientation` methods, and not re-implement them in all of the drivers.

### CODE REVIEW

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

File: test/runtests.jl [hunk 1]
```diff
@@ -3,7 +3,8 @@ using Test, Pkg
 ENV["DATADEPS_ALWAYS_ACCEPT"] = true
 ENV["JULIA_LOG_LEVEL"] = "WARN"
 
-for submodule in ["Utilities/TicToc",
+for submodule in [
+                  "Utilities/TicToc",
```
Comment (kpamnany) [inline test/runtests.jl]: This change looks a bit weird to me, but it's a stylistic choice...

Comment (charleskawczynski) [reply test/runtests.jl]: Ah, yes. It does look a bit funny, but I find this convenient when I want to locally test specific modules (and comment the rest). I can revert if you'd like

Comment (kpamnany) [reply test/runtests.jl]: Don't mind much either way, and your explanation makes sense.

File: experiments/Atmos_GCM/heldsuarez.jl [hunk 2]
```diff
@@ -16,13 +16,15 @@ const p_ground = MSLP
...
-function init_heldsuarez!(state, aux, coords, t)
+function init_heldsuarez!(bl, state, aux, coords, t)
     global p_ground, T_initial
 
     FT = eltype(state)
 
+    # TODO: change to altitude function
     r = norm(coords, 2)
     h = r - FT(planet_radius)
+    # h = altitude(bl.orientation, aux)
```
Comment (kpamnany) [inline experiments/Atmos_GCM/heldsuarez.jl]: Need to add `using CLIMA.Atmos: altitude` for this to work. Or export `altitude` in `orientation.jl`.

Comment (charleskawczynski) [reply experiments/Atmos_GCM/heldsuarez.jl]: I was thinking that I'd do that in a separate PR

**Pull request merged**